Tests/Marketplace/upload_packs.py

```diff
@@ -570,8 +565,65 @@ def add_private_packs_to_index(index_folder_path: str, private_index_path: str):
             update_index_folder(index_folder_path, d.name, d.path)
 
 
+def was_private_pack_updated(private_index_json, public_index_json):
+    """
+    Checks if Content commit hash of each private pack in private and public index files are equal. If there is at
+        least one private pack that has a different content commit hash, it tells us that this pack was updated and
+        should be uploaded. So, an upload flow should NOT be skipped.
+
+    Args:
+        private_index_json (dict) : The private index file.
+        public_index_json (dict) : The public index file.
+
+    Returns:
+        (bool) True if Content commit hash of at least one private pack was changed.
+
+    """
+    id_to_commit_hash_from_public_index = {}
+    for private_pack in public_index_json.get("packs"):
+        pack_id = private_pack.get("id")
+        content_commit_hash = private_pack.get("contentCommitHash", "")
+        id_to_commit_hash_from_public_index[pack_id] = content_commit_hash
+
+    for private_pack in private_index_json.get("packs"):
+        pack_id = private_pack.get("id")
+        content_commit_hash = private_pack.get("contentCommitHash", "")
+        if id_to_commit_hash_from_public_index[pack_id] != content_commit_hash:
+            return True
+    return False
+
+
+def is_there_private_packs_to_upload(public_index_json, private_index_path):
+    """ Checks if there are private packs that should be uploaded.
+    The check compares the private index with the public one to verify if Content commit hash of each private pack in
+    those files (private and public index files) are equal. If there is one private pack that has a different
+    content commit hash, it tells us that this pack was updated and should be uploaded. So, an upload flow should NOT
+    be skipped.
+
+    Args:
+        public_index_json (dict) : The public index file.
+        private_index_path : Path to where the private index is located.
+
+    Returns:
+        (bool) True is there is at least one private pack that should be upload.
+              False otherwise (i.e there are no private packs that should upload)
+
+    """
+    logging.debug("Checking if there are private packs to upload")
+
+    with open(os.path.join(private_index_path, f"{GCPConfig.INDEX_NAME}.json")) as private_index_file:
+        private_index_json = json.load(private_index_file)
```

Comment:
```suggestion
    with open(os.path.join(private_index_path, f"{GCPConfig.INDEX_NAME}.json")) as file:
        private_index_json = json.load(file)
```
Naming here is a bit confusing (index.json vs index.zip) so IMO it's better to change to something generic.